auth-web/src/components/auth/manage-business/AffiliationAction.vue

```diff
@@ -135,16 +135,24 @@ export default defineComponent({
 
     /** Create a business record in LEAR. */
     const createBusinessRecord = async (business: Business): Promise<string> => {
+      const amalgamationTypes = launchdarklyServices.getFlag(LDFlags.SupportedAmalgamationEntities)?.split(' ') || []
       const regTypes = [CorpTypes.SOLE_PROP, CorpTypes.PARTNERSHIP]
       const iaTypes = [CorpTypes.BENEFIT_COMPANY, CorpTypes.COOP, CorpTypes.BC_CCC, CorpTypes.BC_COMPANY,
         CorpTypes.BC_ULC_COMPANY]
 
       let filingResponse = null
-      if (regTypes.includes(business.nameRequest?.legalType)) {
-        filingResponse = await businessStore.createNamedBusiness({ filingType: FilingTypes.REGISTRATION, business })
-      } else if (iaTypes.includes(business.nameRequest?.legalType)) {
-        filingResponse = await businessStore.createNamedBusiness({
-          filingType: FilingTypes.INCORPORATION_APPLICATION, business })
+      // If Incorporation or Registration
+      if (business.nameRequest?.requestActionCd === NrRequestActionCodes.NEW_BUSINESS) {
+        if (regTypes.includes(business.nameRequest?.legalType)) {
+          filingResponse = await businessStore.createNamedBusiness({ filingType: FilingTypes.REGISTRATION, business })
```

Comment:
I'd recommend refactoring this so you call createNamedBusiness once (at the end), and the payload changes accordingly

eg.
`payload = { filingType: FilingTypes.REGISTRATION, business }`
`payload = { filingType: FilingTypes.INCORPORATION_APPLICATION, business }`

`filingResponse = await businessStore.createNamedBusiness(payload)`

it declutters the code a tad